src/nspawn/nspawn.c

```diff
@@ -5417,16 +5415,7 @@ static int cant_be_in_netns(void) {
         if (r < 0)
                 return log_error_errno(r, "Failed to determine peer of udev control socket: %m");
 
-        xsprintf(udev_path, "/proc/" PID_FMT "/ns/net", ucred.pid);
-        r = readlink_malloc(udev_path, &udev_ns);
-        if (r < 0)
-                return log_error_errno(r, "Failed to read network namespace of udev: %m");
-
-        r = readlink_malloc("/proc/self/ns/net", &our_ns);
-        if (r < 0)
-                return log_error_errno(r, "Failed to read our own network namespace: %m");
-
-        if (!streq(our_ns, udev_ns))
+        if (in_same_namespace(ucred.pid, 0, NAMESPACE_NET) <= 0)
```

Comment:
Hm? You mean `if (!in_same_namespace(...))`?

BTW, please handle negative errno.

Reply:
No, I meant to error out either if there's an error or they're not in the same netns.

Reply:
> BTW, please handle negative errno.

I do but I think you mean to handle it with by emitting a separate log line. Did so now.